pvlib/solarposition.py

```diff
@@ -1445,3 +1454,96 @@ def sun_rise_set_transit_geometric(times, latitude, longitude, declination,
     sunset = _local_times_from_hours_since_midnight(times, sunset_hour)
     transit = _local_times_from_hours_since_midnight(times, transit_hour)
     return sunrise, sunset, transit
+
+
+def spencer_mc(times, latitude, longitude):
+    """
+    Calculate the solar position using a python implementation of the
+    Spencer (1972) formulation provided by meteocontrol
+
+    Parameters
+    ----------
+    times : :class:`pandas.DatetimeIndex`
+        Corresponding timestamps, must be localized to the timezone for the
+        ``latitude`` and ``longitude``.
+    latitude : float
+        Latitude in degrees, positive north of equator, negative to south
+    longitude : float
+        Longitude in degrees, positive east of prime meridian, negative to west
+
+    Returns
+    -------
+    DataFrame
+        The DataFrame will have the following columns:
+        zenith (degrees),
+        elevation (degrees),
+        azimuth (degrees),
+        equation_of_time (seconds),
+        eccentricity,
+        declination (degrees).
+
+    References
+    ----------
+    [1] Spencer (1972) can be found in
+    Sen, Zekai. Solar energy fundamentals and modeling techniques:
+     atmosphere, environment, climate change and renewable energy.
+     Springer Science & Business Media, 2008.
+    """
+
+    julians = datetime_to_julian(times)
+    julians_2000 = np.asarray(julians, dtype=np.float) - JULIAN_2000
+
+    lat = np.radians(latitude)
+
+    # Compute fractional year (gamma) in radians
+    gamma = 2 * np.pi * (julians_2000 % JULIAN_YEARS) / JULIAN_YEARS
+    cos_gamma = np.cos(gamma), np.cos(gamma * 2), np.cos(gamma * 3)
+    sin_gamma = np.sin(gamma), np.sin(gamma * 2), np.sin(gamma * 3)
+    day_time = (julians_2000 % 1) * 24
+
+    # Eccentricity: correction factor of the earth's orbit.
+    eccentricity = 1.00011 + 0.034221 * cos_gamma[0]
+    eccentricity += 0.001280 * sin_gamma[0]
+    eccentricity += 0.000719 * cos_gamma[1]
+    eccentricity += 0.000077 * sin_gamma[1]
+
+    declination = np.array(declination_spencer71(times.dayofyear))
+
+    # Equation of time (difference between standard time and solar time).
+    eot = 0.000075 + 0.001868 * cos_gamma[0]
+    eot -= 0.032077 * sin_gamma[0]
+    eot -= 0.014615 * cos_gamma[1]
+    eot -= 0.040849 * sin_gamma[1]
+    eot *= 229.18
```

Comment:
this entire section is a repeat of [`equation_of_time_spencer71`](https://github.com/pvlib/pvlib-python/blob/b20ca5319fd2ea01b0261401d112b70f49146604/pvlib/solarposition.py#L1072-L1079):
```python
    day_angle = _calculate_simple_day_angle(dayofyear)
    # convert from radians to minutes per day = 24[h/day] * 60[min/h] / 2 / pi
    eot = (1440.0 / 2 / np.pi) * (
        0.0000075 +
        0.001868 * np.cos(day_angle) - 0.032077 * np.sin(day_angle) -
        0.014615 * np.cos(2.0 * day_angle) - 0.040849 * np.sin(2.0 * day_angle)
    )
    return eot
```

I would prefer to be [DRY vs WET](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) -- can you pleases consider reusing the existing code functions?

EG: `eot = equation_of_time_spencer71(dayofyear)`

where I suspect that you can use the julian dates you've calculated in place of day of year, or adjust them slightly

see [`_calculate_simple_day_angle`](https://github.com/pvlib/pvlib-python/blob/b20ca5319fd2ea01b0261401d112b70f49146604/pvlib/solarposition.py#L1000-L1014)

Reply:
Right. New implementatino uses the [equation_of_time_spencer71](https://github.com/pvlib/pvlib-python/blob/b20ca5319fd2ea01b0261401d112b70f49146604/pvlib/solarposition.py#L1072-L1079) function. 
Acually I am not really familiar with the Julian Days and Day Of Year usage. I will check it out if there are any ways to use existend code.